core/contracts/financial-templates/implementation/PricelessPositionManager.sol

```diff
@@ -359,14 +357,15 @@ contract PricelessPositionManager is FeePayer, AdministrateeInterface {
         if (positionData.tokensOutstanding.isEqual(numTokens)) {
             amountWithdrawn = _deleteSponsorPosition(msg.sender);
         } else {
-            // Decrease the sponsors position size of collateral and tokens.
-            _removeCollateral(positionData.rawCollateral, collateralRedeemed);
+            // Decrement the sponsor's collateral and global collateral amounts.
+            amountWithdrawn = _decrementCollateralBalancesCheckGCR(positionData, collateralRedeemed);
+
+            // Decrease the sponsors position tokens size. Ensure it is above the min sponsor size.
             FixedPoint.Unsigned memory newTokenCount = positionData.tokensOutstanding.sub(numTokens);
             require(newTokenCount.isGreaterThanOrEqual(minSponsorTokens));
             positionData.tokensOutstanding = newTokenCount;
 
-            // Decrease the contract's collateral and tokens.
-            amountWithdrawn = _removeCollateral(rawTotalPositionCollateral, collateralRedeemed);
```

Comment:
Between the reduction of collateral in the `rawCollateral` of the position and the reduction in the `rawTotalPositionCollateral` here we don't do any checks against the global GCR. Is this something we should check in this context? As a result of using the `_decrementCollateralBalancesCheckGCR` this check is performed between these reductions and thus there is one unit test that fails as a result when redeeming as the only sponsor in the position. @nicholaspai what do you suggest in this case? this basically means that the decrement logic is different in each case.